src/peergos/shared/user/App.java

```diff
@@ -47,9 +47,11 @@ private App(UserContext ctx, Path appDataDirectory) {
 
     @JsMethod
     public static CompletableFuture<App> init(UserContext ctx, String appName) {
-        App app = new App(ctx, Paths.get(ctx.username, APPS_DIR_NAME, appName, DATA_DIR_NAME));
+        Path path = Paths.get(ctx.username == null ? "<guest>" : ctx.username, APPS_DIR_NAME, appName, DATA_DIR_NAME);
```

Comment:
Making up a dir doesn't sit right with me. Maybe App doesn't need it? There is also UserContext.getEntryPath